src/marks/tip.js

```diff
@@ -322,28 +327,86 @@ function getPath(anchor, m, r, width, height) {
   }
 }
 
-function getSources({channels}) {
+function getSourceChannels(index, {channels}, scales) {
+  const {facet, format} = this;
   const sources = {};
+  // Prioritize channels with explicit formats, in the given order.
+  for (const key in format) {
+    if (format[key] === null || format[key] === false) continue;
+    if (key === "fx" || key === "fy") sources[key] = true;
+    else {
+      const source = getSource(channels, key);
+      if (source) sources[key] = source;
+    }
+  }
+  // Then fallback to all other (non-ignored) channels.
   for (const key in channels) {
-    if (ignoreChannels.has(key)) continue;
+    if (key in sources || key in format || ignoreChannels.has(key)) continue;
     const source = getSource(channels, key);
     if (source) sources[key] = source;
   }
+  // And lastly facet channels, but only if this mark is faceted.
+  if (facet) {
+    if (scales.fx && !("fx" in format)) sources.fx = true;
+    if (scales.fy && !("fy" in format)) sources.fy = true;
+  }
   return sources;
 }
 
-function formatPair(c1, c2, i) {
+function formatTitle(i, index, {title}) {
+  return formatDefault(title.value[i], i);
+}
+
+function* formatChannels(i, index, channels, scales, values) {
+  for (const key in channels) {
+    if (key === "fx" || key === "fy") {
+      yield {
+        label: formatLabel(scales, channels, key),
+        value: this.format[key](index[key], i)
+      };
+      continue;
+    }
+    if (key === "x1" && "x2" in channels) continue;
+    if (key === "y1" && "y2" in channels) continue;
+    const channel = channels[key];
+    if (key === "x2" && "x1" in channels) {
+      yield {
+        label: formatPairLabel(scales, channels, "x"),
+        value: formatPair(this.format.x, channels.x1, channel, i)
```

Comment:
I think this might approach not work now, at least if you use a shorthand string for the _x_ or _y_ format, because the name of the format (*x*) does not correspond to the name of the channel (*x1* and *x2*). I could switch to using the *x2* here, but I think supporting *x* is more natural since the same format is used for both channels. 🤔

Reply:
Ah yes this breaks the tipAreaBand, tipAreaStack tests (and maybe others), because format.y is undefined. (Too bad we can't catch those automagically.)

Reply:
I need to run, but here's a fix:

```patch
diff --git a/src/marks/tip.js b/src/marks/tip.js
index 0a271e1d..97fa52bb 100644
--- a/src/marks/tip.js
+++ b/src/marks/tip.js
@@ -372,12 +372,12 @@ function* formatChannels(i, index, channels, scales, values) {
     if (key === "x2" && "x1" in channels) {
       yield {
         label: formatPairLabel(scales, channels, "x"),
-        value: formatPair(this.format.x, channels.x1, channel, i)
+        value: formatPair(this.format.x1 ?? this.format.x, this.format.x2 ?? this.format.x, channels.x1, channel, i)
       };
     } else if (key === "y2" && "y1" in channels) {
       yield {
         label: formatPairLabel(scales, channels, "y"),
-        value: formatPair(this.format.y, channels.y1, channel, i)
+        value: formatPair(this.format.y1 ?? this.format.y, this.format.y2 ?? this.format.y, channels.y1, channel, i)
       };
     } else {
       const value = channel.value[i];
@@ -393,10 +393,10 @@ function* formatChannels(i, index, channels, scales, values) {
   }
 }

-function formatPair(formatValue, c1, c2, i) {
+function formatPair(formatValue1, formatValue2, c1, c2, i) {
   return c2.hint?.length // e.g., stackY’s y1 and y2
-    ? `${formatValue(c2.value[i] - c1.value[i], i)}`
-    : `${formatValue(c1.value[i], i)}–${formatValue(c2.value[i], i)}`;
+    ? `${formatValue1(c2.value[i] - c1.value[i], i)}`
+    : `${formatValue1(c1.value[i], i)}–${formatValue2(c2.value[i], i)}`;
 }

 function formatPairLabel(scales, channels, key) {

```

It seems to work on tipAreaStack and tipBin (I can format separately x1 and x2). I haven't investigated what it gives on tips that show a length/difference instead of an extent (?).

Reply:
I pushed what I think is a more general fix, allowing you to specify a format for the _x_ channel as shorthand for both _x1_ and _x2_ (and same for _y_). This means you can control the order in which paired channels appear, and even suppress one of the two paired channels to just show the other. For example, `format: {y1: false}` will just show the high:

<img width="625" alt="Screenshot 2023-08-25 at 12 52 59 PM" src="https://github.com/observablehq/plot/assets/230541/ffcb7428-8a1c-427e-bd14-21028d8378a4">

And `format: {y: true}` will give _y1_ and _y2_ higher priority:

<img width="640" alt="Screenshot 2023-08-25 at 12 53 37 PM" src="https://github.com/observablehq/plot/assets/230541/c8573f7c-968d-4b11-8c27-f370e73d0b3a">